lib/services/service-scaffold.ts

```diff
@@ -154,39 +183,53 @@ export abstract class ServiceUtilities<T> extends WorkerClient<T> implements Ser
 	 * @returns  Express server app.
 	 */
 	protected get expressApp(): express.Express {
-		if (!ServiceUtilities._expressApp) {
-			// Either MESSAGE_SERVICE_PORT from environment or PORT from Heroku environment
-			const port = process.env.MESSAGE_SERVICE_PORT || process.env.PORT;
+		if (this.instanceExpressApp) {
+			return this.instanceExpressApp;
+		}
+		if (!ServiceScaffold.singletonExpressApp) {
+			// Either PROCBOT_PORT from environment or PORT from Heroku environment
+			const port = process.env.PROCBOT_PORT || process.env.PORT;
```

Comment:
Just realised if you want to run more than one procbot, or if you want to listen to more than one port, you can't. I'd suggest passing in port info as a parameter to make bots explicitly state the port to use.

Reply:
Something about this isn't parsing right, I'm going to get lunch.

Reply:
I've worked out what's wrong.  I've created a hodge-podge of a situation where no one place really knows enough about the express server.
The environment knows which listeners to create.  The listeners know if they need an express server.  The can only be one express server per port.  Using a static variable within service-scaffold is kludgey.  Ick ick ick, time for a rethink.
Looking at the github listener for inspiration I note that it creates its own express server, so can't share its port with any other listener.  This also doesn't feel right.
I think we need to discuss this.